man/builtin.doc

```diff
@@ -7534,6 +7534,50 @@ After = 3,
 SubAtom = xyz.
 \end{code}
 
+The following example splits a string of the form
+<name>=<value> into the name part (an atom) and the value (a string).
+
+\begin{code}
+name_value(String, Name, Value) :-
+    sub_atom(String, Before, _, After, "="),
+    !,
+    sub_atom(String, 0, Before, _, Name),
+    sub_atom(String, _, After, 0, Value).
+\end{code}
+
+This example defines a predicate that inserts a value at a position.
+Note that sub_string/5 is used here instead of sub_atom/5, to avoid
+the overhead of creating atoms compared to creating strinngs.
+\begin{code}
+atom_insert(Str, Val, At, NewStr) :-
+    sub_string(Str, 0, At, A1, S1),
+    sub_string(Str, At, A1, _, S2),
+    atomic_list_concat([S1,Val,S2], NewStr).
+\end{code}
+
+If the \arg{NewString} should be a string, use
+atomics_to-string/2 instead of atomic_list_concat/2:
```

Comment:
Typo `to-string`.  See also comment on the original, which I'd propose to keep.

Reply:
Which "comment on the original"?